assets/blocks/course-list-block/index.js

```diff
@@ -52,3 +53,44 @@ export const registerCourseListBlock = () => {
 		scope: [ 'inserter' ],
 	} );
 };
+
+// Hide the settings which are inherited from the Query Loop block
+// but not applicable to our Course List block.
+const withUnnecessarySettingsHidden = ( BlockEdit ) => {
+	return ( props ) => {
+		if (
+			'core/query' === props.name &&
+			props.isSelected &&
+			props.attributes &&
+			props.attributes.className &&
+			'course-list-block' === props.attributes.className
+		) {
+			setTimeout( () => {
```

Comment:
Can we use @wordpress/dom-ready instead of timeout? or useLayoutEffect, considering it is a react component.

Reply:
I tried with domReady and useLayoutEffect instead of setTimeout, unfortunately didn't work for me as expected in this particular scenario :( So stuck with setTimeout for now

Reply:
I agree with Gabriel about the use of `setTimeout`. It's generally considered to be a dangerous function to use in a React component, because it can cause weird bugs.

Reply:
Removed it [here](https://github.com/Automattic/sensei/pull/5449/commits/a415a78a7c90aaed3a8245d22d20b524819452be), we don't need it with our new approach! ☺️ Can you kindly take a look again? Thanks!